proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -62,101 +62,154 @@ A base JSON schema file for SDL core and proxy will be kept locally and an addit
 * All VehicleData items must be defined in the Policy Table (not just the custom/proprietary items), per recommendation by PM's Core team.
 * Root-Level vehicle data type items such as `rpm` are always `mandatory`:`false` as defined by the RPC Spec for the vehicle data RPCs. If a data item is of type `Struct`, that struct can have mandatory parameters but the struct itself cannot be mandatory for the root-level item.
 
+#### Use cases for vehicle data schema update:
+This section defines SDL behavior in case _vehicle_data_items_ schema is not present in PTU or if data in schema does not match with API/exiting schema. So if in a PTU:
+* _vehicle_data_items_ key does not exist; do not update _vehicle_data_items_ local schema/items.
+* _vehicle_data_items_ exists; replace all local schema vehicle data items with new ones except the standard ones.
+* _vehicle_data_items_ exists but is empty; remove all local schema vehicle data items except the standard ones.
+* _vehicle_data_items_ has item with same name as one of standard vehicle data items, ignore those data items.
+
+*Standard vehicle data items are the ones which are, either present in API or are provided in base schema with SDL core release. Standard vehicle data items are immutable.
+
+
+#### Versioning for Vehicle Data Schema
+Since vehicle data schema will be downloaded as part of policy update, we should not send this data with every update when there is no change in schema. One solution could be that we include the schema in sdl_snapshot when requesting policy update, but that again does not solve the issue as snapshot gets large this way.
+
+Solution to this is to include a schema versioning. This version information will be sent along with PTU when schema update is sent to module in PTU and SDL will persist this value in local policy DB. sdl_snapshot will contain this schema version information when requesting the policy update. Policy server would then compare the schema version received from cloud with what is available on server. Based on this comparison, vehicle data schema update would be sent in policy update response. e.g.
+
+```
+"vehicle_data_schema_version":"00.00.01",
+```
+
+#### Versioning and endpoint for OEM mapping table and to provide HMI a way to read OEM mapping table version
+
+OEM mapping table version would be included in PTU, which SDL will persist in local policy DB. HMI needs to be able to read this value on demand so that it can control if and when to download the OEM mapping table. HMI will utilize the _endpoint_ for OEM mapping table file to download this file using _SystemRequest_ type _OEM_SPECIFIC_. OEM mapping table file would have _endpoint_ key (_service_) as  _oem_mapping_file_.
```

Comment:
```suggestion
OEM mapping table version would be included in PTU, which SDL will persist in local policy DB. HMI needs to be able to read this value on demand so that it can control if and when to download the OEM mapping table. HMI will utilize the _endpoint_ for OEM mapping table file to download this file using _SystemRequest_ type _OEM_SPECIFIC_. OEM mapping table file would have _endpoint_ key (_service_) as _oem_mapping_file_.
```